mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager_internal_test.go.erb

```diff
@@ -0,0 +1,298 @@
+	}
+	shouldNotMatch := map[string]string{
+		// mismatching name
+		"https://www.googleapis.com/compute/v1/projects/imre-test/global/instanceTemplates/example-template-custom": "projects/imre-test/global/instanceTemplates/example-template-custom2",
+		"projects/imre-test/global/instanceTemplates/example-template-custom":                                       "https://www.googleapis.com/compute/v1/projects/imre-test/global/instanceTemplates/example-template-custom2",
+		// matching name, but mismatching uniqueId
+		"https://www.googleapis.com/compute/v1/projects/imre-test/global/instanceTemplates/example-template-custom?uniqueId=123": "projects/imre-test/global/instanceTemplates/example-template-custom?uniqueId=1234",
+		"projects/imre-test/global/instanceTemplates/example-template-custom?uniqueId=123":                                       "https://www.googleapis.com/compute/v1/projects/imre-test/global/instanceTemplates/example-template-custom?uniqueId=1234",
+	}
+	for _, v1 := range shouldAllMatch {
+		for _, v2 := range shouldAllMatch {
+			if !compareSelfLinkRelativePathsIgnoreParams("", v1, v2, nil) {
+				t.Fatalf("compareSelfLinkRelativePathsIgnoreParams did not match (and should have) %v and %v", v1, v2)
+			}
+		}
+	}
+
+	for v1, v2 := range shouldNotMatch {
+		if compareSelfLinkRelativePathsIgnoreParams("", v1, v2, nil) {
+			t.Fatalf("compareSelfLinkRelativePathsIgnoreParams did match (and shouldn't) %v and %v", v1, v2)
+		}
+	}
+}
+
+func TestInstanceGroupManager_convertUniqueId(t *testing.T) {
+	matches := map[string]string{
+		// uniqueId not present (should return the same)
+		"https://www.googleapis.com/compute/v1/projects/imre-test/global/instanceTemplates/example-template-custom": "https://www.googleapis.com/compute/v1/projects/imre-test/global/instanceTemplates/example-template-custom",
+		"projects/imre-test/global/instanceTemplates/example-template-custom":                                       "projects/imre-test/global/instanceTemplates/example-template-custom",
+		// uniqueId present (should return the last component replaced)
+		"https://www.googleapis.com/compute/v1/projects/imre-test/global/instanceTemplates/example-template-custom?uniqueId=123": "https://www.googleapis.com/compute/v1/projects/imre-test/global/instanceTemplates/123",
+		"projects/imre-test/global/instanceTemplates/example-template-custom?uniqueId=123":                                       "projects/imre-test/global/instanceTemplates/123",
+		"tf-test-igm-8amncgtq22?uniqueId=8361222501423044003":                                                                    "8361222501423044003",
+	}
+	for input, expected := range matches {
+		actual := ConvertToUniqueIdWhenPresent(input)
+		if actual != expected {
+			t.Fatalf("invalid return value by ConvertToUniqueIdWhenPresent for input %v; expected: %v, actual: %v", input, expected, actual)
+		}
+	}
+}
+
+func TestFlattenStatefulPolicyStatefulExternalIps(t *testing.T) {
```

Comment:
note: just flagging I am only reviewing from this point in the file onwards, let me know if I'm missing anything.

Reply:
Thanks - you're right to do that - the other tests in the file aren't changed but the diff is large just because of the file extension change.